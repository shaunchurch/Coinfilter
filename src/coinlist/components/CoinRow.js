// @flow
import React, { Component } from 'react';
import NumberFormat from 'react-number-format';

import type { Coin } from '../coinlist.types';

type Props = {
  coin: Coin
};

class CoinRow extends Component<Props> {
  render() {
    const { coin } = this.props;
    return (
      <li>
        <img
          src={`https://files.coinmarketcap.com/static/img/coins/32x32/${coin.id}.png`}
        />
        {coin.name} -
        <strong>cap:</strong>
        <NumberFormat
          value={coin.market_cap_usd}
          displayType={'text'}
          thousandSeparator={true}
          prefix={'$'}
        />
        <strong>vol:</strong>
        <NumberFormat
          value={coin['24h_volume_usd']}
          displayType={'text'}
          thousandSeparator={true}
          prefix={'$'}
        />
        <strong>coin:</strong>
        <NumberFormat
          value={coin.price_usd}
          displayType={'text'}
          thousandSeparator={true}
          prefix={'$'}
        />
      </li>
    );
  }
}

export default CoinRow;
