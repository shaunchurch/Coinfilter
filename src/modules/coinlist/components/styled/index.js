import styled from 'styled-components';

export const Row = styled.li`
  display: flex;
  flex-direction: row;
  flex-wrap: no-wrap;
  margin: 8px 0;
  padding: 8px 0;
  color: #999;
  transition: color .15s ease-in-out, background-color .15s ease-in-out;
  &:hover {
    color: #333;
    background-color: #f9f9f9;
  }
`;

export const Cell = styled.div`
  display: flex;
  min-width: 180px;
  max-width: 100%;
  overflow: hidden;
  white-space: nowrap;
  align-items: center;
  text-align: right;
  justify-content: flex-end;
  font-family: 'menlo', 'monaco', 'courier', monospace;
`;

export const Price = Cell.extend`
  width: 110px;
  color: ${props => (props.moon ? props.theme.green : props.theme.red)};
`;

export const Logo = styled.div`width: 40px;`;

export const Rank = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 40px;
  color: #ccc;
`;
