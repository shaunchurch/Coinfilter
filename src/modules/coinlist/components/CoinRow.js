// @flow
import React, { Component } from 'react';
import NumberFormat from 'react-number-format';
import { Row, Rank, Logo, Price, Cell } from './styled';

import type { Coin } from '../coinlist.types';

type Props = {
  coin: Coin
};

class CoinRow extends Component<Props> {
  render() {
    const { coin } = this.props;

    let moon = true;
    if (coin.percent_change_1h < 0) {
      moon = false;
    }
    return (
      <Row>
        <Rank>
          {coin.rank}
        </Rank>
        <Logo width={30}>
          <img
            src={`https://files.coinmarketcap.com/static/img/coins/32x32/${coin.id}.png`}
          />
        </Logo>
        <Price moon={moon}>
          <NumberFormat
            value={coin.price_usd}
            displayType={'text'}
            thousandSeparator={true}
            prefix={'$'}
          />
        </Price>
        <Cell>
          <NumberFormat
            value={coin.market_cap_usd}
            displayType={'text'}
            thousandSeparator={true}
            prefix={'$'}
          />
        </Cell>
        <Cell>
          <NumberFormat
            value={coin['24h_volume_usd']}
            displayType={'text'}
            thousandSeparator={true}
            prefix={'$'}
          />
        </Cell>
      </Row>
    );
  }
}

export default CoinRow;
