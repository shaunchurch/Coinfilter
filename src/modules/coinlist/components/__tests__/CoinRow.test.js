import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CoinRow from '../CoinRow';
import CoinJSON from '../../../../testutils/data/Coin.json';
import { Price } from '../styled';

describe('<CoinRow />', () => {
  it('should render a coin row positive', () => {
    const wrapper = shallow(<CoinRow coin={CoinJSON[0]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render a coin row negative', () => {
    const wrapper = shallow(<CoinRow coin={CoinJSON[1]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render in the correct style when moon', () => {
    const wrapper = shallow(<Price moon={true} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render <Price /> in the correct style when no moon', () => {
    const wrapper = shallow(<Price moon={false} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
