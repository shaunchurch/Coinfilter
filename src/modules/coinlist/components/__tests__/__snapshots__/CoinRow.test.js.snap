// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoinRow /> should render <Price /> in the correct style when no moon 1`] = `
<div
  className="sc-htpNat iSValt"
/>
`;

exports[`<CoinRow /> should render a coin row negative 1`] = `
<styled.li>
  <styled.div>
    2
  </styled.div>
  <styled.div
    width={30}
  >
    <img
      src="https://files.coinmarketcap.com/static/img/coins/32x32/ethereum.png"
    />
  </styled.div>
  <styled.div
    moon={true}
  >
    <NumberFormat
      allowNegative={true}
      decimalSeparator="."
      displayType="text"
      isAllowed={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      prefix="$"
      suffix=""
      thousandSeparator={true}
      type="text"
      value={333.882}
    />
  </styled.div>
  <styled.div>
    <NumberFormat
      allowNegative={true}
      decimalSeparator="."
      displayType="text"
      isAllowed={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      prefix="$"
      suffix=""
      thousandSeparator={true}
      type="text"
      value={31472781162}
    />
  </styled.div>
  <styled.div>
    <NumberFormat
      allowNegative={true}
      decimalSeparator="."
      displayType="text"
      isAllowed={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      prefix="$"
      suffix=""
      thousandSeparator={true}
      type="text"
      value={514181000}
    />
  </styled.div>
</styled.li>
`;

exports[`<CoinRow /> should render a coin row positive 1`] = `
<styled.li>
  <styled.div>
    1
  </styled.div>
  <styled.div
    width={30}
  >
    <img
      src="https://files.coinmarketcap.com/static/img/coins/32x32/bitcoin.png"
    />
  </styled.div>
  <styled.div
    moon={false}
  >
    <NumberFormat
      allowNegative={true}
      decimalSeparator="."
      displayType="text"
      isAllowed={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      prefix="$"
      suffix=""
      thousandSeparator={true}
      type="text"
      value={4352.4}
    />
  </styled.div>
  <styled.div>
    <NumberFormat
      allowNegative={true}
      decimalSeparator="."
      displayType="text"
      isAllowed={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      prefix="$"
      suffix=""
      thousandSeparator={true}
      type="text"
      value={71929882019}
    />
  </styled.div>
  <styled.div>
    <NumberFormat
      allowNegative={true}
      decimalSeparator="."
      displayType="text"
      isAllowed={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      prefix="$"
      suffix=""
      thousandSeparator={true}
      type="text"
      value={1511610000}
    />
  </styled.div>
</styled.li>
`;

exports[`<CoinRow /> should render in the correct style when moon 1`] = `
<div
  className="sc-htpNat iSValt"
/>
`;
